Updated documentation to include PersistKeysToDbContext (#22003)
dotnet/AspNetCore.Docs: aspnetcore/security/data-protection/configuration/overview.md

> [!WARNING]
> If you change the key persistence location, the system no longer automatically encrypts keys at rest, since it doesn't know whether DPAPI is an appropriate encryption mechanism.

## PersistKeysToDbContext

To store keys in a database using EntityFramework, configure the system with the [Microsoft.AspNetCore.DataProtection.EntityFrameworkCore](https://www.nuget.org/packages/Microsoft.AspNetCore.DataProtection.EntityFrameworkCore/) package:

```csharp
public void ConfigureServices(IServiceCollection services)
{
    services.AddDataProtection()
        .PersistKeysToDbContext<DbContext>()
}
```

> This will store the keys used in the configured database. The database context being used must adhere to `IDataProtectionKeyContext` which exposes the property `DataProtectionKeys` 

Reviewer: ```suggestion
The preceding code stores the keys in the configured database. The database context being used must implement `IDataProtectionKeyContext`.  `IDataProtectionKeyContext` exposes the property `DataProtectionKeys` 
```

```csharp
public DbSet<DataProtectionKey> DataProtectionKeys { get; set; }
```

> This property represents the table in which the keys will be stored. The definition of this type [can be found here](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.dataprotection.entityframeworkcore.dataprotectionkey?view=aspnetcore-5.0). This table will need to be created either manually or via your DbContext Migrations.

Reviewer: ```suggestion
This property represents the table in which the keys are stored. Create the table manually or with `DbContext` Migrations. See <xref:Microsoft.AspNetCore.DataProtection.EntityFrameworkCore.DataProtectionKey> for more information.
```
